### PR TITLE
Create merge_two_lists.py that implements merging of two sorted linked lists

### PR DESCRIPTION
**Pull request by maldz3**
…d lists

### **Describe your change:**
Implemented a new algorithm that merges two sorted linked lists into one sorted linked list
Includes:
function that creates a linked list using the Node class from a Python array, 
function that creates a string representation of a linked list (1->2->3),
function that merges two lists into one.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by maldz3** [comment]
Please let me know if there are any changes or additions I can make.

**Comment by maldz3** [comment]
I'd appreciate some feedback and a code review for me pr, thanks.

**Review by xcodz-dot — APPROVED** [review]
@dhruvmanila, check this out

Also well done

**Review by cclauss — CHANGES_REQUESTED** [review]
I am struggling with why we need all this code to deliver the equivalent of `list_one + list_two`

File: data_structures/linked_list/merge_two_lists.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    num_list.sort()
+    current = head = Node(num_list[0])
+    for i in range(1, len(num_list)):
+        current.next = Node(num_list[i])
+        current = current.next
+    return head
+
+
+def str_linked_list(head: Node) -> str:
```
[the remark this answers is not in the document]
Comment (maldz3) [reply data_structures/linked_list/merge_two_lists.py:34]: -Thank you both for responding, I appreciate the feedback!  The main purpose of the algorithm is to merge two linked lists, however I added the create_linked_list function for ease of testing and so users can quickly create linked lists, not sure if that answers your question?
-I can definitely change that line to def __str__()
-I'm not really sure how and where to add the __iter__() method to my algorithm

[the remark this answers is not in the document]
Comment (xcodz-dot) [reply data_structures/linked_list/merge_two_lists.py:34]: You can try this link to understand about iter and implementing it

https://www.geeksforgeeks.org/python-__iter__-__next__-converting-object-iterator/

[the remark this answers is not in the document]
Comment (maldz3) [reply data_structures/linked_list/merge_two_lists.py]: Thanks for the link.  Looks like I will have to add a LinkedList class in order to have` __str__` and `__iter__`, which means all the functions will have to be changed accordingly.  Should I go ahead and make those changes?

[the remark this answers is not in the document]
Comment (xcodz-dot) [reply data_structures/linked_list/merge_two_lists.py]: Noop I suggest you to look at this file and make changes there (its not a good idea to keep two linked list classes) Also, then delete your file and modify changes into this one

https://github.com/TheAlgorithms/Python/blob/9c6080a6fc31ae7737535ba305de20a5cc0805b5/data_structures/linked_list/__init__.py

[the remark this answers is not in the document]
Comment (cclauss) [reply data_structures/linked_list/merge_two_lists.py]: Don’t copy that one...  it does not have `__iter__()`

[the remark this answers is not in the document]
Comment (maldz3) [reply data_structures/linked_list/merge_two_lists.py]: I could include this file in my file and use the builtin methods if that works?
https://github.com/TheAlgorithms/Python/blob/master/data_structures/linked_list/singly_linked_list.py

[the remark this answers is not in the document]
Comment (xcodz-dot) [reply data_structures/linked_list/merge_two_lists.py]: good idea (You can try it but wait for @cclauss)

**Comment by maldz3** [comment]
@cclauss I've simplified the file by importing the Node and LinkedList classes from https://github.com/TheAlgorithms/Python/blob/master/data_structures/linked_list/singly_linked_list.py
Are there any other changes you would like me to make?  Thanks!

**Comment by xcodz-dot** [comment]
I can not tell much but wait for @cclauss, I do not find importing relevent in this case

**Review by cclauss — APPROVED** [review]

**Pull request merged**